tests/phpunit/tests/link/getAdjacentPost.php

```diff
@@ -622,6 +622,236 @@ public function test_get_adjacent_post_with_identical_dates() {
 		$this->assertEquals( $post_ids[3], $next->ID );
 	}
 
+	/**
+	 * Test that deterministic ID fallback is applied when WHERE filter doesn't modify the clause.
+	 *
+	 * @ticket 64390
+	 */
+	public function test_get_adjacent_post_identical_dates_applies_deterministic_where_when_filter_unmodified() {
+		$identical_date = '2024-01-01 12:00:00';
+
+		// Create posts with identical dates but different IDs.
+		$post_ids = array();
+		for ( $i = 1; $i <= 5; $i++ ) {
+			$post_ids[] = self::factory()->post->create(
+				array(
+					'post_title' => "Post $i",
+					'post_date'  => $identical_date,
+				)
+			);
+		}
+
+		// Add a filter that doesn't modify the WHERE clause (returns unchanged).
+		add_filter(
+			'get_next_post_where',
+			static function ( $where ) {
+				// Return unchanged - deterministic fallback should be applied.
+				return $where;
+			}
+		);
+
+		// Test navigation from the middle post (ID: 3rd post).
+		$current_post_id = $post_ids[2]; // 3rd post
+		$this->go_to( get_permalink( $current_post_id ) );
+
+		// Next post should be the 4th post (higher ID, same date) - deterministic.
+		$next = get_adjacent_post( false, '', false );
+		$this->assertInstanceOf( 'WP_Post', $next );
+		$this->assertEquals( $post_ids[3], $next->ID );
```

Comment:
```suggestion
		$this->assertSame( $post_ids[3], $next->ID );
```

:1234: Looks to apply elsewhere so I won't repeat myself.

Reply:
👍🏻